Singer.API/Controllers/DataControllerBase.cs

```diff
@@ -0,0 +1,180 @@
+
+         // get the search results of the database query
+         var result = await DatabaseService.GetAsync(
+           filter: filter,
+           orderer: orderByLambda,
+           sortDirection: direction,
+           page: pageIndex,
+           itemsPerPage: pageSize);
+
+
+         var requestPath = HttpContext.Request.Path;
+         var nextPage = (pageIndex * pageSize) + result.Size >= result.TotalCount
+            ? null
+            : $"{requestPath}?PageIndex={pageIndex + pageSize}&Size={pageSize}";
+
+         // create object that holds the paginated elements
+         var page = new PaginationDTO<TDTO>
+         {
+            Items = result.Items,
+            Size = result.Items.Count,
+            PageIndex = pageIndex,
+            CurrentPageUrl = $"{requestPath}?{HttpContext.Request.QueryString.ToString()}",
+            NextPageUrl = nextPage,
+            PreviousPageUrl = pageIndex == 0
+               ? null
+               : $"{requestPath}?PageIndex={pageIndex - pageSize}&Size={pageSize}",
+            TotalSize = result.TotalCount
+         };
+
+         return Ok(page);
+      }
+
+      /// <summary>
+      /// The GET method to get a single entity from the database.
+      /// </summary>
+      /// <param name="id">Id of the element to return.</param>
+      /// <returns>The entity in the database that has the given id.</returns>
+      [HttpGet("{id}")]
+      [ProducesResponseType(StatusCodes.Status200OK)]
+      [ProducesResponseType(StatusCodes.Status404NotFound)]
+      [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+      public virtual async Task<ActionResult<CareUserDTO>> GetOne(Guid id)
+      {
+         var dto = await DatabaseService.GetOneAsync(id);
+         return Ok(dto);
+      }
+
+      #endregion get
+
+      #region put
+
+      /// <summary>
+      /// Updates a single entity in the database.
+      /// </summary>
+      /// <param name="id">The id of the entity to update.</param>
+      /// <param name="dto">The new value of the entity.</param>
+      /// <returns></returns>
+      [HttpPut("{id}")]
+      [ProducesResponseType(StatusCodes.Status200OK)]
+      [ProducesResponseType(StatusCodes.Status404NotFound)]
+      [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+      public async Task<ActionResult> UpdateUser(Guid id, TDTO dto)
+      {
+         var result = await DatabaseService.UpdateAsync(id, dto);
+         return Ok(result);
+      }
+
+      #endregion put
+
+      #region delete
+
+      /// <summary>
+      /// Deletes a single entity from the database.
+      /// </summary>
+      /// <param name="id">The id of the entity to remove.</param>
+      /// <returns></returns>
+      [HttpDelete("{id}")]
+      [ProducesResponseType(StatusCodes.Status204NoContent)]
+      [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+      public async Task<ActionResult> DeleteUser(Guid id)
```

Comment:
should be `DeleteItem`, not user specific

Reply:
fixed it